testsuite/tests/asmcomp/func_sections.ml

```diff
@@ -0,0 +1,57 @@
+(* TEST
+   * native
+*)
+
+(* Test for anonymous functions which result in a mangled symbol *)
+let f4 list =
+  List.map (fun s -> String.length s) list
+
+let test1 () =
+  f4 ["a";"asfda";"afda"]
+
+(* Test for jump tables*)
+
+let g1 s = s^"*"
+let g2 s = "*"^s
+let g3 s = "*"^s^"*"
+
+let f5 = function
+  | 1 -> g1 "a"
+  | 2 -> g2 "b"
+  | 3 -> g3 "c"
+  | 4 -> g1 "d"
+  | 5 -> g2 "e"
+  | 6 -> g3 "f"
+  | _ -> "x"
+
+let test2 () =
+  let list =    [f5 5;
+                f5 7;
+                f5 15;
+                f5 26]
+  in
+  ignore list
+
+let iter = 1_000
+
+let f0 x = x - 7;
+[@@inline never]
+
+let f1 x = x + iter
+[@@inline never]
+
+let f2 x = f1(x)
+[@@inline never]
+
+let f3 x = f2(x)*f0(x)
+[@@inline never]
+
+let test3 () =
+  f3 iter
+
+
+let () =
+  ignore (test1 ());
+  ignore (test2 ());
+  ignore (test3 ());
+  ()
```

Comment:
I don't see what that test is supposed to test.
I suppose that you have to inspect the generated assembly to see if it works as expected. If it is the case, maybe some script that messes with the result can be added. Like linking using some weird linker script.
Don't bother if this is too complex.

Reply:
Yes, this test is useless without some script. I checked the results manually by looking at the assembly output (produced with -S) to see that it has named text sections. It is awkward to do with ocamltest but possible. I've just push a commit that does it c2bcd9a2e82ba1e1ed5fb12cf95f31b6117312ca . As you can see, there is a bit of a hack in func_sections.run to check the generated assembly, and it requires a few new variables to be added in ocamltest. 

I also used a linker script that reorders the functions and then checked using "nm" that the order of function symbols is as expected. I have a script for it but it is tricky because it needs to check that the tools for binary inspection are available. I don't think we need it in the testsuite, as function reordering is not part of the compiler.

Reply:
I realized that I can simplify the conditions of this test case using the configure option. The updated version of the test is in commit 798c4f6769007002e5a302739fd204153d06d5bf.